examples/ellbvp_linfe/ellbvp_linfe_demo.cc

```diff
@@ -0,0 +1,363 @@
+/**
+ * @file
+ * @brief Solution of general second-order elliptic boundary value problem with
+ * linear finite elements
+ * @author Ralf Hiptmair
+ * @date   January 2019
+ * @copyright MIT License
+ */
+
+#include <fstream>
+#include <iomanip>
+
+#include <lf/assemble/assemble.h>
+#include <lf/geometry/geometry.h>
+#include <lf/io/io.h>
+#include <lf/lagrfe/lagrfe.h>
+#include <lf/mesh/hybrid2d/hybrid2d.h>
+#include <lf/mesh/utils/utils.h>
+#include <lf/refinement/refinement.h>
+#include "lf/mesh/test_utils/test_meshes.h"
+
+int main(int argc, const char** argv) {
+  std::cout << "\t LehrFEM++ Demonstration Code " << std::endl;
+  std::cout << "\t Solution of general second-order elliptic\n"
+            << "\t boundary value problem by means of linear\n"
+            << "\t Lagrangian finite element discretization" << std::endl;
+
+  // abbreviations for types
+  using size_type = lf::base::size_type;
+  using glb_idx_t = lf::assemble::glb_idx_t;
+  using coord_t = Eigen::Vector2d;
+
+  // ======================================================================
+  // Prep stage: provide all coefficient functions mainly through lambda
+  //             functions and derived MeshFunctions.
+  // ======================================================================
+
+  // Coefficients:
+
+  // 2x2 diffusion tensor A(x)
+  std::function<Eigen::Matrix<double, 2, 2>(Eigen::Vector2d)> alpha =
```

Comment:
Sorry if it wasn't clear. I just wanted to point out that `auto alpha = ...` is not the same as `std::function<...> alpha = ...`! The latter *converts* the lambda object into a std::function which uses type erasure internally. Strictly speaking, the std::function conversion is not necessary in the code (and it also slows down the code a bit, but since this is definitely not the bottle neck this effect is less relevant)

Reply:
I was not aware of the difference: I have removed all `std::function` declarations.